src/test/java/io/kestra/storage/gcs/GcsStorageTest.java

```diff
@@ -162,4 +170,112 @@ void deleteByPrefixNoResult() throws Exception {
         List<URI> deleted = storageInterface.deleteByPrefix(tenantId, new URI("/" + prefix + "/storage/"));
         assertThat(deleted.size(), is(0));
     }
+
+    @Test
+    void list() throws Exception {
+        String prefix = IdUtils.create();
+        String tenantId = IdUtils.create();
+        URL resource = GcsFileAttributes.class.getClassLoader().getResource("application.yml");
+
+        List<String> path = Arrays.asList(
+            "/" + prefix + "/storage/root.yml",
+            "/" + prefix + "/storage/level1/1.yml",
+            "/" + prefix + "/storage/level1/level2/1.yml",
+            "/" + prefix + "/storage/another/1.yml"
+        );
+        path.forEach(throwConsumer(s -> this.putFile(tenantId, resource, s)));
+
+        List<FileAttributes> list = storageInterface.list(tenantId, new URI("/" + prefix + "/storage"));
+
+        assertThat(list.stream().map(FileAttributes::getFileName).toList(), containsInAnyOrder("root.yml", "level1", "another"));
```

Comment:
Please verify that it's size is 3 so that you ensure you don't retrieve subdirectories' files

Reply:
containsInAnyOrder already check that it only contains the input:
>Creates an order agnostic matcher for Iterables that matches when a single pass over the examined Iterable yields a series of items, each logically equal to one item anywhere in the specified items. For a positive match, the examined iterable must be of the same length as the number of specified items.

Reply:
didn't know, ty :+1: